scripts/store.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Store new assets into IPFS.
+ * Look for directories under `./assets`, only store assets without `uri` in `state.${ENV}.json` file.
+ * Update `state.${ENV}.json` file after storage.
+ */
+
+const fs = require("fs")
+const path = require("path")
+const { NFTStorage, File } = require("nft.storage")
+
+const client = new NFTStorage({
+  token: process.env.PINNING_SERVICE_KEY,
+})
+
+const env = process.env.ENV
+
+async function main() {
+  // get path to assets
+  const assetsDirectory = path.join(__dirname, "..", "assets")
+
+  // read current asset state or initialize
+  const assetsStatePath = path.join(assetsDirectory, `state.${env}.json`)
+
+  let assetsState
+  try {
+    assetsState = require(assetsStatePath)
+  } catch (err) {
+    assetsState = {}
+  }
+
+  // find assets to be stored
+  const assetPaths = fs
+    .readdirSync(assetsDirectory, { withFileTypes: true })
+    .filter((dirent) => {
+      // filter out directories
+      if (!dirent.isDirectory()) {
+        return false
+      }
+
+      // filter out assets that have not been stored
+      if (assetsState[dirent.name] && assetsState[dirent.name].uri) {
+        return false
+      }
+
+      return true
+    })
+    .map((dirent) => dirent.name)
+
+  if (assetPaths.length > 0) {
+    // store all new assets
+    await Promise.all(
+      assetPaths.map(async (asset) => {
+        // asset directory
+        const diretory = path.join(__dirname, "..", "assets", asset)
+
+        // read in metadata
+        const assetJson = require(path.join(diretory, "metadata.json"))
+
+        // store on IPFS via client
+        console.log(`Storing asset ${asset}...`)
+        const { url } = await client.store({
+          ...assetJson,
+          image: new File(
+            [await fs.promises.readFile(path.join(diretory, assetJson.image))],
+            assetJson.image,
+            { type: "image/jpg" }
```

Comment:
is avatar always saved `jpeg`? or just in the Draft PR;     does someone have png or gif

Reply:
Good point, it can be any image. I Will update it accordingly.